### PR TITLE
vardahoth/LatestChanges

### PR DESCRIPTION
**Pull request by vardahoth**
Refactored getAddress().toString(), filled in skeleton matching methods for Address, fixed CompareContactPartsTest street address tests and apartment address tests

### CODE REVIEW

File: src/com/cft/contactmerge/CompareContactParts.java [hunk 1]
```diff
@@ -6,9 +6,14 @@ public static AnswerType doNamesMatch(String firstName1, String lastName1, Strin
...
     }
 
-    public static AnswerType doAddressesMatch(String address1, String address2)
+    public static AnswerType doStreetAddressesMatch(Contact contact, String otherStreetAddress)
     {
-        return AddressMatchSelector.getStreetAddressMatchResultDefaultComparisons(address1, address2);
+        return contact.getAddress().isStreetAddressMatch(otherStreetAddress);
+    }
+
+    public static AnswerType doApartmentAddressesMatch(Contact contact, String otherApartmentAddress)
```
Comment (mbilodeau6) [inline src/com/cft/contactmerge/CompareContactParts.java]: The fact that this is returning null makes me suspect that there is no test or there is a fake test. This is fine to do on a branch but I wouldn't merge this to main if we are still planning to follow the rules we have in the Wiki.

File: src/com/cft/contactmerge/contact/Address.java [hunk 2]
```diff
@@ -78,41 +80,81 @@ public void setZip(String zip) {
     /*******************************************************************************************************************
      *************************************************** Sub-Methods ***************************************************
      *******************************************************************************************************************/
+    @Override
```
Comment (mbilodeau6) [inline src/com/cft/contactmerge/contact/Address.java]: Thumbs up. (Git need emoticons).

File: src/com/cft/contactmerge/tests/CompareContactPartsTest.java [hunk 3]
```diff
@@ -84,118 +85,207 @@ void doNamesMatch_No_Apostrophe_NameCombined() { // Note: In the future, we may
      ******************************************* Compare Street Address Tests ******************************************
      *******************************************************************************************************************/
     @Test
-    void doAddressesMatch_Yes_IgnoreCase() {
-        assertEquals(AnswerType.yes, CompareContactParts.doAddressesMatch("123 main Street", "123 Main street"));
+    void doStreetAddressesMatch_Yes_IgnoreCase() {
+        Contact contactOne = new Contact();
+
+        contactOne.getAddress().setStreetAddress("123 main Street");
+        assertEquals(AnswerType.yes, CompareContactParts.doStreetAddressesMatch(contactOne, "123 Main street"));
```
Comment (mbilodeau6) [inline src/com/cft/contactmerge/tests/CompareContactPartsTest.java]: I think you are overloading this test class, aren't you? It is set up to test CompareContactParts (the static methods) but you are now adding tests that are creating/using full Contacts. I was expecting those kinds of tests to be in the ContactTest class. If it isn't clear what I mean, I can show you what I was thinking on Monday. My concern is that your tests/classes are getting more dependent on one another which will make troubleshooting and maintenance more challenging.

File: src/com/cft/contactmerge/CompareContactParts.java (hunk 1 above)
Comment (vardahoth) [reply src/com/cft/contactmerge/CompareContactParts.java]: it's a skeleton method, it hasn't been implemented yet, and there are tests for it.

File: src/com/cft/contactmerge/tests/CompareContactPartsTest.java (hunk 3 above)
Comment (vardahoth) [reply src/com/cft/contactmerge/tests/CompareContactPartsTest.java]: We'll talk more about this. But I'm going to go ahead and implement the streetAddressMatch(). Worst case scenario, if I have to decouple, I will only have to change the tests.

**Pull request merged**